pyproject.toml

```diff
@@ -87,3 +81,93 @@ task-tags = ["Example", "todo", "TODO", "FIXME"]
 
 [tool.codespell]
 ignore-words-list = 'edn,juxt'
+
+[build-system]
+requires = ["setuptools>=65", "setuptools_scm[toml]>=7.1"]
+
+[tool.setuptools]
+packages = ["rocky", "octopoes", "keiko", "boefjes", "mula", "bytes"]
+
+[tool.setuptools_scm]
+write_to = "_version.py"
+
+[project]
+name = "openkat"
+version = "0.0.1.dev1"
+description = "OpenKAT coordination package"
+authors = [
+    { name = "MinVWS", email = "maintainer@openkat.nl" }
+]
+license = { text = "EUPL" }
+readme = "README.rst"
+requires-python = ">=3.8"
+
+[project.optional-dependencies]
```

Comment:
I think we should not pin the versions in pyproject.toml, because we already do that in the requirements.txt. That way we can easily upgrade everything to the latest version by using `pip-compile -U` or upgrade a specific package with `pip-compile -P`. We can also make it clear that we don't want to upgrade a certain package to newer version, for example specifying `django<5` for rocky.

Reply:
I think we should still pin packages on major versions for stable projects (Django 4.x), and minor versions for projects that have more breakage (FastAPI 0.95.x). This is what the [caret (^) in Poetry](https://python-poetry.org/docs/dependency-specification/#caret-requirements) does. I don't think the [PEP-508](https://peps.python.org/pep-0508/) that `pip-compile` supports in `pyproject.toml` support this, but we can write version ranges to to the same.

Reply:
Preventing breakages is already done by pinning the versions in requirements.txt. Pinning minor versions in pyproject.toml only make sense if the intentation is to stay on those versions, which in not the case of FastAPI as those minor versions don't receive security updates as far as I know. I think it is even more important to regularly upgrade to the latest version because of the breakage, because if we are a lot of behind and there is a security issue it's harder to upgrade to the latest version with the fix because more things need to be fixed.

Reply:
Disagree, if running `pip-compile -U` often results in breaking the build then we simply won't run it as often. I think it would be good to often upgrade packages to minor versions within compatible ranges, and manually go through breaking changes (like FastAPI updates) on a regular basis when there is time and expertise to fix breaking changes.

Reply:
We already have automated updating dependencies with dependabot and dependabot is not going to run less often because things break. I don't think it is a good idea to skip updates for things that are likely to break. I would rather see a dependabot PR with failing CI. The alternatives is having to remember that we have to do this manually which we wil llikely forget until it is urgent because dependabot notifies about a security issue.